app/controllers/works_controller.rb

```diff
@@ -0,0 +1,100 @@
+class WorksController < ApplicationController
+  before_action :set_work, only: [:show, :edit, :update, :destroy, :upvote]
+
+  def index
+    @works = Work.all
+    @albums = Work.sorted_media("album")
+    @books = Work.sorted_media("book")
+    @movies = Work.sorted_media("movie")
+  end
+
+  def show
+  end
+
+  def new
+    @work = Work.new
+  end
+
+  def edit
+  end
+
+  def create
+    @work = Work.new(work_params)
+    if @work.save 
+      flash[:success] = "You created a new #{@work.category}"
+      redirect_to works_path(@work.id)
+      return
+    else
+      flash.now[:error] = "Work was not created, please try again!"
+      render :new
+      return
+    end
+  end
+
+  def update
+    # @work = Work.find_by(id: params[:id])
+    if @work.nil?
+      flash.now[:error] = "Work was not found"
+      redirect_to works_path
+      return 
+    elsif @work.update(work_params)
+      flash[:success] = "#{@work.title} was successfully updated."
+      redirect_to works_path
+      return
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    if @work.nil?
+      flash.now[:error] = "Work was not found"
+      redirect_to works_path
+      return
+    else
+      @work.destroy 
+      flash.now[:notice] = "You have successfully deleted #{work.title}"
```

Comment:
This line produces error: undefined local variable or method `work' for #<WorksController:0x00007f8e7e6bbb10>
Did you mean?  @work

```suggestion
      flash.now[:notice] = "You have successfully deleted #{@work.title}"
```